### PR TITLE
Add a section on exceptions to the Daml intro

### PR DESCRIPTION
**Pull request by cocreature**
### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by cocreature** [comment]
You might want to review this commit by commit. The first one only renumbers things.

File: docs/source/daml/intro/8_Exceptions.rst [hunk 1]
```diff
@@ -0,0 +1,137 @@
+.. Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+8 Exception Handling
+====================
+
+The default behavior in Daml is to abort the transaction on any error
+and rollback all changes that have happened until then. However, this
```
Comment (sofiafaro-da) [inline docs/source/daml/intro/8_Exceptions.rst:8]: I looked it up and it turns out English is weird: "rollback" is the noun form, and "roll back" is the verb form. (It's just like "login" vs "log in".)

```suggestion
and roll back all changes that have happened until then. However, this
```

File: docs/source/daml/intro/8_Exceptions.rst [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+be invasive for operations where aborting the transacton is often the
+desired behavior, e.g., changing division to return ``Either`` or an
+``Option`` to handle division by zero would be a very invasive change
+and many callsites might not want to handle the error case explicitly.
+Second, and more importantly, this approach does not allow rolling
+back ledger actions that have happened up to this point: If a contract
+got created before we hit the error, there is no way to undo this.
+
+Exceptions provide a way to handle certain types of errors in a
+non-invasive way and allow to rollback parts of the transaction that
```
Comment (sofiafaro-da) [inline docs/source/daml/intro/8_Exceptions.rst:25]:
```suggestion
non-invasive way and to roll back parts of the transaction that
```

**Review by sofiafaro-da — APPROVED** [review]
Awesome work, thank you!

File: docs/source/daml/intro/8_Exceptions.rst (hunk 1 above)
Comment (cocreature) [reply docs/source/daml/intro/8_Exceptions.rst:8]: > I looked it up and it turns out English is weird:

who could have thought :slightly_smiling_face:

**Pull request merged**